### PR TITLE
LPD-34782 Fix com.liferay.portal.security.auth.test.AuthenticatedSessionManagerUtilTest

### PR DESCRIPTION
**Pull request by ChrisKian**
Forwarded from https://github.com/liferay-appsec/liferay-portal/pull/1915 with minor SF change, see 8317e4ec89a9be746fd76f9617e13ff9ab08f300.

Original message:

> Related issue: [LPD-34873](https://liferay.atlassian.net/browse/LPD-34873)
> 
> After a new impl of [LogoutPreAction#Run](https://github.com/liferay-appsec/liferay-portal/blob/8aecdeea188c7659808f4893f1adab1ebbbe48d9/modules/apps/click-to-chat/click-to-chat-web/src/main/java/com/liferay/click/to/chat/web/internal/events/LogoutPreAction.java#L30), the test started failing because of NPE. I added a ThemeDisplay for the test to properly work.

### CODE REVIEW

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
ci:test:security

**Comment by manuelecastro** [comment]
#### Test fixed successfully! :heavy_check_mark: 
see: [Testray results](https://testray.liferay.com/#/project/34754/routines/34756/build/72555567/case-result/72619897)

File: modules/apps/portal-security/portal-security-auth-test/src/testIntegration/java/com/liferay/portal/security/auth/test/AuthenticatedSessionManagerUtilTest.java [hunk 1]
```diff
@@ -129,6 +134,16 @@ public void testRememberMeLogout() throws Exception {
 			CookiesConstants.CONSENT_TYPE_FUNCTIONAL, cookie,
 			mockHttpServletRequest, mockHttpServletResponse);
 
+		ThemeDisplay themeDisplay = ThemeDisplayFactory.create();
+
+		Group group = GroupTestUtil.addGroup();
+
+		themeDisplay.setCompany(CompanyTestUtil.addCompany());
```
Comment (stian-sigvartsen) [inline modules/apps/portal-security/portal-security-auth-test/src/testIntegration/java/com/liferay/portal/security/auth/test/AuthenticatedSessionManagerUtilTest.java:140]: @manuelecastro is it necessary to create a whole new Company for this test? It bears a high cost.

Comment (manuelecastro) [reply modules/apps/portal-security/portal-security-auth-test/src/testIntegration/java/com/liferay/portal/security/auth/test/AuthenticatedSessionManagerUtilTest.java:140]: @stian-sigvartsen I added as [LogoutPreAction](https://github.com/liferay-appsec/liferay-portal/blob/8aecdeea188c7659808f4893f1adab1ebbbe48d9/modules/apps/click-to-chat/click-to-chat-web/src/main/java/com/liferay/click/to/chat/web/internal/events/LogoutPreAction.java#L40) requests `themeDisplay.getCompanyId()`. Is there a way we can escape this?

Comment (stian-sigvartsen) [reply modules/apps/portal-security/portal-security-auth-test/src/testIntegration/java/com/liferay/portal/security/auth/test/AuthenticatedSessionManagerUtilTest.java:140]: I think so. I will push a commit

File: modules/apps/portal-security/portal-security-auth-test/src/testIntegration/java/com/liferay/portal/security/auth/test/AuthenticatedSessionManagerUtilTest.java [hunk 2]
```diff
@@ -129,6 +131,15 @@ public void testRememberMeLogout() throws Exception {
 			CookiesConstants.CONSENT_TYPE_FUNCTIONAL, cookie,
 			mockHttpServletRequest, mockHttpServletResponse);
 
+		ThemeDisplay themeDisplay = ThemeDisplayFactory.create();
+
+		themeDisplay.setCompany(
+			_companyLocalService.getCompany(TestPropsValues.getCompanyId()));
+		themeDisplay.setSiteGroupId(TestPropsValues.getGroupId());
```
Comment (manuelecastro) [inline modules/apps/portal-security/portal-security-auth-test/src/testIntegration/java/com/liferay/portal/security/auth/test/AuthenticatedSessionManagerUtilTest.java:138]: @stian-sigvartsen This makes a lot more sense, I didn't know about `TestPropsValues`, thank you!

**Comment by manuelecastro** [comment]
@stian-sigvartsen @ChrisKian this is passing locally, should we run the security suite anyway?

**Comment by manuelecastro** [comment]
Actually, the test is passing but there are still errors being thrown in the portal log.
see:
```2024-08-28 21:45:14.236 ERROR [com.liferay.portal.security.auth.test-executor-thread][LogoutPreAction:52] null
java.lang.NullPointerException: null
	at com.liferay.commerce.internal.events.LogoutPreAction.run(LogoutPreAction.java:38) [bundleFile:?]
	at com.liferay.portal.kernel.events.Action.processLifecycleEvent(Action.java:25) [portal-kernel.jar:?]
	at com.liferay.portal.events.EventsProcessorUtil.process(EventsProcessorUtil.java:83) [portal-impl.jar:?]
	at com.liferay.portal.events.EventsProcessorUtil.process(EventsProcessorUtil.java:43) [portal-impl.jar:?]
	at com.liferay.portal.security.auth.session.AuthenticatedSessionManagerUtil.logout(AuthenticatedSessionManagerUtil.java:250) [portal-impl.jar:?]
	at com.liferay.portal.security.auth.test.AuthenticatedSessionManagerUtilTest.testRememberMeLogout(AuthenticatedSessionManagerUtilTest.java:143) [bundleFile:?]
```

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
ci:test:security

**Comment by manuelecastro** [comment]
#### Test is passing :heavy_check_mark: 
see: [Testray result](https://testray.liferay.com/#/project/34754/routines/34756/build/74814541/case-result/74939493)
cc: @ChrisKian @stian-sigvartsen

**Review by stian-sigvartsen — APPROVED** [review]
LGTM!

**Comment by stian-sigvartsen** [comment]
ci:forward

**Pull request closed without merging**